docs/contributing.rst

```diff
@@ -0,0 +1,161 @@
+Contributing
+------------
+
+Thank you for your interest in contributing! We welcome all contributions no matter their size. Please read along to learn how to get started.
+
+
+Setting the stage
+~~~~~~~~~~~~~~~~~
+
+First, we need to clone the <PROJECT_NAME> repository:
+
+.. code:: sh
+
+    $ git clone https://github.com/ethereum/<REPO_NAME>.git
+
+
+.. include:: /fragments/virtualenv_explainer.rst
+
+After we have activated our virtual environment, we need to install all dependencies that are needed to run, develop, and test.
+This is as easy as navigating to the <REPO_NAME> directory and running:
+
+.. code:: sh
+
+    pip install -e ".[dev]"
+
+
+Running the tests
+~~~~~~~~~~~~~~~~~
+
+A great way to explore the code base is to run the tests.
+
+We can run all tests with:
+
+.. code:: sh
+
+    pytest tests
+
+However, you may just want to run a subset instead, like:
+
+.. code:: sh
+
+    pytest tests/core/padding-utils/test_padding.py
+
+
+We can also install ``tox`` to run the full test suite which also covers things like testing the code against different Python versions, linting etc.
+
+It is important to understand that each Pull Request must pass the full test suite as part of the CI check, hence it is often convenient to have ``tox`` installed locally as well.
+
+
+Code Style
+~~~~~~~~~~
+
+When multiple people are working on the same body of code, it is important that they write code that conforms to a similar style. It often doesn't matter as much which style, but rather that they conform to one style.
+
+To ensure your contribution conforms to the style being used in this project, we encourage you to read our `style guide <https://github.com/pipermerriam/ethereum-dev-tactical-manual/blob/main/style-guide.md>`_.
+
+
+Type Hints
+~~~~~~~~~~
+
+This codebase uses `type hints <https://www.python.org/dev/peps/pep-0484/>`_. Type hints make it easy to prevent certain types of bugs, enable richer tooling, and enhance the documentation, making the code easier to follow.
```

Comment:
This is aspirationally true, but I don't think all of our repos have type hinting yet. Not sure how/if we want to highlight that, or just remove from libraries that don't have it yet.

Reply:
Removed the section for now. Can re-add once all libs have mypy as a dep, so at least new code can be typechecked.